snap/local/start-mysqlrouter-exporter.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -eo pipefail # Exit on error
+
+EXPORTER_OPTS="--collect.metadata.status \
+--collect.route.connections.byte_from_server \
+--collect.route.connections.byte_to_server \
+--collect.route.connections.time_started \
+--collect.route.connections.time_connected_to_server \
+--collect.route.connections.time_last_sent_to_server \
+--collect.route.connections.time_received_from_server"
+EXPORTER_PATH="/usr/bin/mysqlrouter_exporter"
+
+if [ -n "$SNAP" ]; then
+    EXPORTER_PATH="${SNAP}${EXPORTER_PATH}"
+    MYSQLROUTER_EXPORTER_USER="$(snapctl get mysqlrouter-exporter.user)"
+    MYSQLROUTER_EXPORTER_PASS="$(snapctl get mysqlrouter-exporter.password)"
+    MYSQLROUTER_EXPORTER_URL="$(snapctl get mysqlrouter-exporter.url)"
+    TLS_OPTS="--skip-tls-verify"
+
+    if [ -z "$MYSQLROUTER_EXPORTER_URL" ]; then
+        echo "mysqlrouter-exporter.url must be set"
+        exit 1
+    fi
+
+    if [[ -z "$MYSQLROUTER_EXPORTER_USER" || -z "$MYSQLROUTER_EXPORTER_PASS" ]]; then
+        echo "mysqlrouter-exporter.user and mysqlrouter-exporter.password must be set"
+        exit 1
+    fi
+
+    # For security measures, daemons should not be run as sudo.
+    # Execute mysqlrouter-exporter as the non-sudo user: snap-daemon.
+    exec "$SNAP"/usr/bin/setpriv \
+        --clear-groups \
+        --reuid snap_daemon \
+        --regid snap_daemon -- \
+        env MYSQLROUTER_EXPORTER_URL="${MYSQLROUTER_EXPORTER_URL}" \
+        MYSQLROUTER_EXPORTER_USER="${MYSQLROUTER_EXPORTER_USER}" \
+        MYSQLROUTER_EXPORTER_PASS="${MYSQLROUTER_EXPORTER_PASS}" \
+        "$EXPORTER_PATH" $(echo "$EXPORTER_OPTS") $(echo "$TLS_OPTS")
```

Comment:
since $EXPORTER_OPTS is not a command, running `$(EXPORTER_OPTS)` will try to execute the options as a command. therefore, echo just prints the opts which is what we want (so they get passed into the `$EXPORTER_PATH` command

Reply:
it does indeed. changed in [6cd064f](https://github.com/canonical/charmed-mysql-snap/pull/41/commits/6cd064fea62d8f7cbfc4700ab8acb12970dbe08a)

Reply:
actually, it does not work with multiline options (where we use \). changed back to use echo in [8cf0c36](https://github.com/canonical/charmed-mysql-snap/pull/41/commits/8cf0c3693ae5d16b8d513730c7c5d1796791fdbe)